grails-app/services/au/org/ala/biocollect/merit/CommonService.groovy

```diff
@@ -40,4 +42,19 @@ class CommonService {
         }
         return featuresMap as JSON
     }
+
+    /**
+     * copies grails parameters but ignores a predefined list of keys
+     * @param params
+     * @return
+     */
+    Map parseParams(Map params){
+        Map result = [:]
+        params.each { key, value ->
```

Comment:
Have a look at the ```collect``` and ```findResults``` methods - they're a bit nicer for this sort of operation, where you are building a collection while iterating over another.